components/core/src/clp/clp/FileCompressor.cpp

```
@@ -177,22 +171,54 @@
-    m_file_reader.close();
-
     Profiler::stop_continuous_measurement<Profiler::ContinuousMeasurementIndex::ParseLogFile>();
     LOG_CONTINUOUS_MEASUREMENT(Profiler::ContinuousMeasurementIndex::ParseLogFile)
     PROFILER_SPDLOG_INFO("Done parsing {}", file_name)
 
     return succeeded;
 }
 
+auto FileCompressor::parse_and_encode(
+        size_t target_data_size_of_dicts,
+        streaming_archive::writer::Archive::UserConfig& archive_user_config,
```

cppcheck

[error] Exception thrown in function declared not to throw exceptions.

(throwInNoexceptFunction)